### PR TITLE
feat: /api/v0/dns

### PR DESCRIPTION
**Pull request by hacdias**
Ref.: #904, https://github.com/ipfs/js-ipfs-api/pull/665

On Node.js we use `dns.resolveTxt` to get the `dnslink` path. On the browser, we use the public gateway.

### CODE REVIEW

**Comment by hacdias** [comment]
image

**Comment by hacdias** [comment]
@diasdavid some tests will fail since it doesn't use the js-ipfs-api from https://github.com/ipfs/js-ipfs-api/pull/665

**Comment by hacdias** [comment]
And this one too is done @diasdavid 🚀

**Comment by hacdias** [comment]
Ping @diasdavid

File: test/cli/dns.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  before(function () {
+    this.timeout(60 * 1000)
+    ipfs = thing.ipfs
+  })
+
+  it('resolve ipfs.io dns', function () {
+    this.timeout(60 * 1000)
+
+    return ipfs('dns ipfs.io').then((res) => {
+      console.log(res)
```
Comment (diasdavid) [inline test/cli/dns.js]: @hacdias not needed, right?

Comment (hacdias) [reply test/cli/dns.js]: Done!

**Comment by hacdias** [comment]
Fixed @diasdavid

**Comment by hacdias** [comment]
Fixed @diasdavid! Was using `window` variable which is not available on WebWorkers 😄

**Comment by hacdias** [comment]
Is there anything missing here @diasdavid?

**Pull request merged**